libfreerdp/gdi/graphics.c

```diff
@@ -250,48 +280,62 @@ void gdi_Glyph_Free(rdpContext* context, rdpGlyph* glyph)
 	}
 }
 
-void gdi_Glyph_Draw(rdpContext* context, rdpGlyph* glyph, int x, int y)
+BOOL gdi_Glyph_Draw(rdpContext* context, rdpGlyph* glyph, int x, int y)
 {
 	gdiGlyph* gdi_glyph;
 	rdpGdi* gdi = context->gdi;
 
 	gdi_glyph = (gdiGlyph*) glyph;
 
-	gdi_BitBlt(gdi->drawing->hdc, x, y, gdi_glyph->bitmap->width,
-			gdi_glyph->bitmap->height, gdi_glyph->hdc, 0, 0, GDI_DSPDxax);
+	if (gdi_BitBlt(gdi->drawing->hdc, x, y, gdi_glyph->bitmap->width,
+			gdi_glyph->bitmap->height, gdi_glyph->hdc, 0, 0, GDI_DSPDxax) != 0)
+		return FALSE;
+	return TRUE;
 }
 
-void gdi_Glyph_BeginDraw(rdpContext* context, int x, int y, int width, int height, UINT32 bgcolor, UINT32 fgcolor, BOOL fOpRedundant)
+BOOL gdi_Glyph_BeginDraw(rdpContext* context, int x, int y, int width, int height, UINT32 bgcolor, UINT32 fgcolor, BOOL fOpRedundant)
 {
 	GDI_RECT rect;
 	HGDI_BRUSH brush;
 	rdpGdi* gdi = context->gdi;
+	int ret = 1;
 
 	/* TODO: handle fOpRedundant! See xf_Glyph_BeginDraw() */
 
 	bgcolor = freerdp_convert_gdi_order_color(bgcolor, gdi->srcBpp, gdi->format, gdi->palette);
 	fgcolor = freerdp_convert_gdi_order_color(fgcolor, gdi->srcBpp, gdi->format, gdi->palette);
 
-	gdi_CRgnToRect(x, y, width, height, &rect);
-
 	brush = gdi_CreateSolidBrush(fgcolor);
-	gdi_FillRect(gdi->drawing->hdc, &rect, brush);
+	if (!brush)
+	{
+		goto out_fail;
+		return FALSE;
+	}
+
+	gdi_CRgnToRect(x, y, width, height, &rect);
+	ret = gdi_FillRect(gdi->drawing->hdc, &rect, brush);
 	gdi_DeleteObject((HGDIOBJECT) brush);
 
+out_fail:
 	gdi->textColor = gdi_SetTextColor(gdi->drawing->hdc, bgcolor);
+
+	if (ret != 0)
```

Comment:
OMG ... Now I see why the original code did not use any return value checking ... obviously  because nobody really knew the meaning.
Thanks, you're doing a heroic job here!